### PR TITLE
Add saga prompt & detection on route generator.

### LINKED ISSUE #761: Bug - generate route registers non-existent saga
ref [#744 (comment)](https://github.com/mxstbr/react-boilerplate/pull/744#issuecomment-233851411)

Generating a route to a container with a reducer, but without a saga registers a saga in the route. This results in a broken route/container that is not trivial to fix.

Repro steps:
- Create a new container `npm run generate container` Accept defaults _except_ saga.
image

- Create a route `npm run generate route`
image

- Start server `npm start`
image

- Edit ./app/routes.js to remove reference to saga
image
image

- Restart server (no error), open `http://localhost:3000/test` Get blank screen and console error
image

### PR DESCRIPTION
**Pull request by kraffslol**
This adds a prompt to the route generator asking you if you want to use sagas for the route you're generating. It automatically detects if the component has a sagas file and sets the default accordingly.

Fixes #761.

### CODE REVIEW

File: internals/generators/route/index.js [hunk 1]
```diff
@@ -39,6 +50,11 @@ module.exports = {
 
       return 'path is required';
     },
+  }, {
+    type: 'confirm',
+    name: 'useSagas',
+    default: sagasExists(componentName) ? true : false,
+    message: 'Do you want to use sagas?',
```
Comment (gihrig) [inline internals/generators/route/index.js]: How about changing message to:
> Do you want sagas for asynchronous flows? (e.g. fetching data)

This is the sagas prompt from the container generator.

Comment (kraffslol) [inline internals/generators/route/index.js]: Sounds good. I'll change the message.

**Comment by kraffslol** [comment]
The generator will now only prompt you if it can find a related sagas file to your component. If it doesn't it assumes that no saga is being used and adds a route without the saga lines.

**Comment by mxstbr** [comment]
a) This looks great, thanks!

b) Not a fan of the prompt, if a saga exists add it to the route automatically. Why would we ask the user again?

**Comment by gihrig** [comment]
Tested:
Container: no reducer, no saga -> no saga prompt -> app works 
Container: no reducer, has saga -> answer No to saga prompt -> app works 
Container: no reducer, has saga -> answer Yes to saga prompt -> app works 
Container: has reducer, no saga -> no saga prompt -> app works 
Container: has reducer, has saga -> answer No to saga prompt -> app works
Container: has reducer, has saga -> answer Yes to saga prompt -> app works

Works for me 😃 

Nice work @kraffslol, @mxstbr  recommend merge 💯

**Comment by kraffslol** [comment]
@mxstbr I haven't found a way to pass a variable to the handlebars template without adding a prompt entry. There doesn't seem to be a way to pass a variable through the action directly.

The alternative I see to bypass the prompt is creating another template i.e. `routeWithReducerNoSagas.hbs` and doing the `sagasExists` call in the actions method, like reducerExists currently does. I having trouble deciding which solution is the best.

**Comment by mxstbr** [comment]
I would strongly prefer not to have a prompt, since I cannot think of a case where the user wants to code split the route but not the saga. The saga will not load and the user will be heavily confused…

**Comment by kraffslol** [comment]
Are you okay with the approach of adding another template then?

**Comment by mxstbr** [comment]
Yeah, I guess that's fine if there's no other way… /cc @amwmedia who is the creator of plop, maybe he has some insights here!

**Comment by amwmedia** [comment]
@kraffslol @mxstbr Can one of you boil down the question for me there? Something about passing variables to handlebars?

**Comment by amwmedia** [comment]
I THINK the answer is that yes you can pass data into handlebars by just adding another property to the `data` object within your actions function. The `data` object that is passed into the actions function is the same, mutable, object that handlebars receives. So you can change the user's answers, or add data to it as needed.

**Comment by kraffslol** [comment]
@amwmedia Thank you! It's just what we needed.

**Comment by amwmedia** [comment]
@kraffslol glad I could help

**Comment by mxstbr** [comment]
Awesome, thanks so much for doing this @kraffslol!! :tada: (and also @amwmedia for chiming in obviously!)

**Comment by gihrig** [comment]
@kraffslol Tested all 4 combos of reducers and sagas(none, R, S, RS) works perfectly.

Awesome stuff, very helpful! ❤️

**Pull request merged**